core/amber/src/main/python/core/runnables/network_receiver.py

```diff
@@ -25,7 +25,15 @@ def __init__(self, shared_queue: InternalQueue, host: str, port: int):
 
         # register the data handler to deserialize data messages.
         @logger.catch(reraise=True)
-        def data_handler(command: bytes, table: Table):
+        def data_handler(command: bytes, table: Table) -> int:
+            """
+            Data handler for deserializing data messages
+
+            :param command:
+            :param table:
+            :return: number of batches in internal queue belonging to this command's sender actor.
+                this is used for calculating sender credits
```

Comment:
just return a `credit`. how to determine the credit is up to implementation.

Reply:
Done, thanks